rom/dev/README.md

```diff
@@ -668,6 +668,42 @@ Following is the sequence of steps that are performed to download the firmware i
 
 See Firmware [Image Validation Process](#firmware-image-validation-process).
 
+### Derivation of the hash chain for Stable Identity
+
+Stable Identity calls for a secret that remains stable across firmware updates, but which can ratchet forward when major firmware vulnerabilities are fixed. Caliptra ROM implements this feature in terms of a "hash chain".
+
+The hash chain is initialized from the LDevID CDI during cold-boot. The hash chain length is inversely related to the firmware's SVN. Each step in the chain is an SVN-unique key. The key for SVN X can be obtained by applying a one-way cryptographic operation to the key for SVN X+1. In this manner, firmware with a given SVN can wield keys bound to its SVN or older, but cannot wield keys bound to newer SVNs.
+
+To comply with FIPS, the one-way cryptographic operation used to compute keys is an SP 800-108 KDF.
+
+When the hash chain is initialized at cold-boot, it is bound to the lifecycle state, debug-locked state, and the firmware's "epoch" from the image header. This ensures that across lifecycle or debug state transtions, or across intentional epoch changes, the keys in the hash chain will change.
+
+Across update-resets, ROM tracks the minimum SVN that has run since cold-boot. It ensures that the chain's length always corresponds to that minimum SVN. The hash chain can only be shortened (and thereby give access to newer SVNs' keys) by cold-booting into firmware with a newer SVN and re-initializing the chain.
+
+#### Cold-boot
+
+ROM initializes a hash chain for the firmware. LDevID CDI in Key Vault Slot6 is used as an HMAC Key, and the data is a fixed string. The resultant MAC is stored in Slot 2.
+
+    `HashChainContext = lifecycle state || debug_locked state || firmware epoch`
+
+    `hmac512_kdf(KvSlot6, label: b"hash_chain", context: HashChainContext, KvSlot2)`
+
+    Loop (MAX_FIRMWARE_SVN - (current firmware SVN)) times:
+
+        `hmac512_kdf(KvSlot2, label: b"", context: None, KvSlot2)`
```

Comment:
I think we should always put something in the label, even though it is probably not strictly necessary here.

Reply:
Done, though I think what we really want is just to ensure that any time a key is branched off from the chain, the label is different than what was used to extend it (in this case just make sure it's non-empty).

Have gone with "si_init" and "si_extend" for the two hash chain operations.